### PR TITLE
Updating README with basic overview

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -4,24 +4,15 @@ edx-when
...
-------------------------
-
-The ``README.rst`` file should then provide an overview of the code in this
-repository, including the main components and useful entry points for starting
-to understand the code in more detail.
-
-Documentation
--------------
-
-The full documentation is at https://edx-when.readthedocs.org.
```
Comment (Dillon-Dumesnil) [inline README.rst]: This is not a real link so definitely shouldn't be in here.

File: README.rst [hunk 2]
```diff
@@ -4,24 +4,15 @@ edx-when
...
+Overview
+--------
+
+edx-when is designed to be the central source of dates for the LMS.
+It is written to by Studio when a Course is published
+(via https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/course_date_signals/handlers.py)
+and then the LMS reads from it in several locations.
+This repo contains start, end, and due dates for Courses and offers the
+functionality to have both absolute and relative dates.
```
Comment (robrap) [inline README.rst]: Thanks so much @Dillon-Dumesnil.  This makes it sound like an IDA with its own db.  Is it an app deployed to LMS and Studio and sharing the same DB?  If so, can we add that clarification?

File: README.rst (hunk 1 above)
Comment (cpennington) [reply README.rst]: We should get edx-when publishing to RTD, though, so that that can actually be a real link.

File: README.rst (hunk 2 above)
Comment (Dillon-Dumesnil) [reply README.rst]: Ah yes, it is installed inside of edx-platform and does share that same database. I will put up a quick follow-up. Good call out!

File: README.rst (hunk 1 above)
Comment (Dillon-Dumesnil) [reply README.rst]: Agreed. We will add it back in once we have that set up.

File: README.rst (hunk 2 above)
Comment (Dillon-Dumesnil) [reply README.rst]: Please let me know what you think @robrap https://github.com/edx/edx-when/pull/41

**Pull request merged**